Sources/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -403,19 +403,26 @@ class PurchasesOrchestrator {
         } catch StoreKitError.userCancelled {
             return (
                 transaction: nil,
-                customerInfo: try await self.syncPurchases(receiptRefreshPolicy: .always, isRestore: false),
+                customerInfo: try await self.customerInfoManager.customerInfo(appUserID: self.appUserID,
+                                                                              fetchPolicy: .cachedOrFetched),
                 userCancelled: true
             )
         } catch {
             throw ErrorUtils.purchasesError(withStoreKitError: error)
         }
 
-        let (userCancelled, customerInfoIfSynced, sk2Transaction) = try await self.storeKit2TransactionListener
+        let (userCancelled, sk2Transaction) = try await self.storeKit2TransactionListener
```

Comment:
don't we already know whether this is a cancellation from the lines before this?

Reply:
The line before catches `StoreKitError.userCancelled`. This comes from `Product.PurchaseResult.userCancelled`. It's unclear when StoreKit would throw one versus returning the other, but we need to handle both.